### PR TITLE
feat: encryption and decryption

### PR DESCRIPTION
**Pull request by Dudleyneedham**
## fixes https://github.com/KILTprotocol/ticket/issues/891

Adding functionality to encryption and decryption of the keys and credentials.

Changing to the redux store.
Changing to the services.

## How to test:

Create a credential and keys to store.

## Checklist:

- [ ] I have verified that the code works
- [ ] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [ ] I have [left the code in a better state](https://deviq.com/boy-scout-rule/)
- [ ] I have documented the changes (where applicable)

### CODE REVIEW

File: src/state/PersistentStore.ts [hunk 1]
```diff
@@ -77,8 +79,11 @@ class PersistentStore {
       uiState: UiState.Store.serialize(),
       wallet: Wallet.Store.serialize(state.wallet),
     }
-
-    return JSON.stringify(obj)
+    const encryptedState = encryption(
+      JSON.stringify(obj),
+      '0x000102030405060708090A0B0C0D0E0F101112131415161718191A1B1C1D1E1F'
```
Comment (tjwelde) [inline src/state/PersistentStore.ts:84]: This can be any string. not necessarily a hex one

File: src/utils/Encryption/Encryption.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+export function encryption(
+  message: string,
+  secret: CryptoInput
+): EncryptedSymmetricString {
+  return encryptSymmetricAsStr(message, secret)
+}
+
+export function decryption(data: string, secret: CryptoInput): string | null {
+  return decryptSymmetricStr(JSON.parse(data), secret)
+}
```
Comment (tjwelde) [inline src/utils/Encryption/Encryption.ts:17]: This whole file seems unnecessary. Do you need it for future work?

**Review by tjwelde — CHANGES_REQUESTED** [review]

File: src/utils/Encryption/Encryption.ts (hunk 2 above)
Comment (Dudleyneedham) [reply src/utils/Encryption/Encryption.ts:17]: I would say for future work it seems necessary. I removed the encryption modal already but kept the `Encryption.ts`.

**Comment by Dudleyneedham** [comment]
Need to add - https://github.com/dchest/scrypt-async-js/ - in order to produce a key to access the local storage. 

This is done with a plain text password and salt. 

Storing the salt also in the local storage separate from the encrypted data. 

First we check if the salt is in the local storage, if not generate a salt and store. Using the salt and password to create the key needed to encrypt and decrypt the local storage.

**Review by tjwelde — APPROVED** [review]

**Pull request merged**